homeassistant/components/microbees/config_flow.py

```diff
@@ -0,0 +1,83 @@
+"""Config flow for microBees integration."""
+from collections.abc import Mapping
+import logging
+from typing import Any
+
+import microBeesPy
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_TOKEN
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import aiohttp_client, config_entry_oauth2_flow
+
+from .const import DOMAIN
+
+VERSION = 1
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class OAuth2FlowHandler(
+    config_entry_oauth2_flow.AbstractOAuth2FlowHandler, domain=DOMAIN
+):
+    """Handle a config flow for microBees."""
+
+    DOMAIN = DOMAIN
+
+    @property
+    def logger(self) -> logging.Logger:
+        """Return logger."""
+        return logging.getLogger(__name__)
+
+    reauth_entry: config_entries.ConfigEntry | None = None
+
+    @property
+    def extra_authorize_data(self) -> dict[str, Any]:
+        """Extra data that needs to be appended to the authorize url."""
+        scopes = ["read", "write"]
+        return {"scope": " ".join(scopes)}
+
+    async def async_oauth_create_entry(self, data: dict) -> FlowResult:
+        """Create an oauth config entry or update existing entry for reauth."""
+
+        microbees = microBeesPy.microbees.MicroBees(
+            session=aiohttp_client.async_get_clientsession(self.hass),
+            token=data[CONF_TOKEN][CONF_ACCESS_TOKEN],
+        )
+
+        try:
+            current_user = await microbees.getMyProfile()
+        except microBeesPy.microbees.MicroBeesException:
+            return self.async_abort(reason="token_invalid")
+        except Exception:  # pylint: disable=broad-except
+            return self.async_abort(reason="unknown")
+
+        if not self.reauth_entry:
+            await self.async_set_unique_id(current_user.id)
+            self._abort_if_unique_id_configured()
+            return self.async_create_entry(
+                title=f"{current_user.firstName} {current_user.lastName} - {current_user.username}",
+                data=data,
+            )
+        if self.reauth_entry.unique_id == current_user.id:
+            self.hass.config_entries.async_update_entry(
+                self.reauth_entry, data={**self.reauth_entry.data, **data}
```

Comment:
You're not storing other data, so might as well just 
```suggestion
                self.reauth_entry, data=data
```